src/main/java/de/uni_tuebingen/ub/nppm/servlet/LoginServlet.java

```diff
@@ -15,50 +15,59 @@
-
-        if (login == null || login.isEmpty() || !BenutzerDB.hasLogin(login)) {
-            response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Benutzer existiert nicht");
-        } else {
-            Benutzer benutzer = BenutzerDB.getByLoginAktiv(login);
-            String saltString = benutzer.getSalt();
-            if (saltString == null || saltString.isEmpty()) {
-                response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Die Sicherheit der Datenbank wurde verbessert. Das Password muss neu gesetzt werden. <a href=\"" + Utils.getBaseUrl(request) + "/forgotPassword\">Neuen Link generieren</a>");
-            } else {
-                byte[] saltBytes = SaltHash.Base64StringToBytes(saltString);
-                String passwordSalted = SaltHash.GenerateHash(password, AuthHelper.getPasswordHashingAlgorithm(), saltBytes);
-                if (!passwordSalted.equals(benutzer.getPassword())) {
-                    response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Ungültiges Passwort.");
-                } else {
-                    // Falls Session vorhanden, löschen
-                    HttpSession session = request.getSession();
-                    if (session != null) {
-                        session.invalidate();
-                    }
-
-                    // Neue Session erzeugen
-                    session = request.getSession(true);
-                    session.setAttribute("BenutzerID", benutzer.getID());
-                    session.setAttribute("GruppeID", benutzer.getGruppe().getID());
-                    session.setAttribute("Benutzername", benutzer.getLogin());
-                    session.setAttribute("Administrator", benutzer.isAdmin());
-                    session.setAttribute("Gast", benutzer.isGast());
-                    session.setAttribute("Sprache", benutzer.getSprache());
-                    session.setMaxInactiveInterval(AuthHelper.getSessionTimeout());
+   protected void processLoginAction(HttpServletRequest request, HttpServletResponse response) throws Exception {
+    String login = request.getParameter("username");
+    String password = request.getParameter("password");
 
-                    // Weiterleiten
-                    if ((Boolean) session.getAttribute("Gast")) {
-                        response.sendRedirect("gast/startseite");
-                    } else {
-                        response.sendRedirect("einzelbeleg");
-                    }
-                }
-            }
-        }
+    if (login == null || login.isEmpty() || !BenutzerDB.hasLogin(login)) {
+        response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Benutzer existiert nicht");
+        return;
+    }
+
+    Benutzer benutzer = BenutzerDB.getByLogin(login);
+
+    if (benutzer == null || !benutzer.isAktiv()) {
+        response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Zugriff nicht erlaubt, Ihr Administrator muss Sie auf aktiv schalten !");
```

Comment:
There should be different messages for when the user is null or when he is not active.

Reply:
done:  look at commit : Login Servlet message